EC-84: Revise Configuration, Customers, eContent, Login and Logging (#5)
ezsystems/commerce-documentation: docs/guide/landing_page/landing_page_api.md

# Landing Page API

## Last viewed products

> The products are recorded using a JavaScript call. In order to use it on other pages as well you have to place this code in your template:

Reviewer: ```suggestion
The products are recorded by means of a JavaScript call. To use it on other pages as well, place the following code in your template:
```

`{{ render_hinclude( controller( ``"SilversolutionsEshopBundle:Catalog:storeProductAsLastViewed"``, {``"sku"``: catalogElement.sku} ) ) }}`

> After each call the cache for the last viewed product slider is purged.

Reviewer: ```suggestion
After each call, the cache for the last viewed product slider is purged.
```

> By default the view is rendered by an ESI block and cached per user.

Reviewer: ```suggestion
By default, the view is rendered by an ESI block and cached per user.
```

> When the customer visits a new product the list is purged and regenerated the next time it is displayed. 

Reviewer: ```suggestion
When the customer visits a new product, the list is purged and regenerated the next time it is displayed. 
```

Products are stored in the session.

### Displaying products

``` html+twig
{{ render_esi(
   controller(
      'SilversolutionsEshopBundle:EzFlow:showLastViewedProducts',
       { }
   ))
}}
```

The controller is able to render a different template if required (parameter template).
`SilversolutionsEshopBundle:Catalog:slider.html.twig` is used by default. 

> The caching strategy can be defined in the config file. Since it is dynamic, you should use `vary: cookie`.

Reviewer: Does "it is" stand for the caching strategy, or the config file? 
```suggestion
The caching strategy can be defined in the config file. It is dynamic, therefore use `vary: cookie`.
```

``` yaml
silver_eshop.default.http_cache:
        last_viewed_products:
            max_age: 36000
            vary: cookie
```

You can also configure the number of stored products (10 by default):

``` yaml
silver_eshop.default.last_viewed_products_in_session_limit: 10
```

## Bestsellers

> The Bestseller block is rendered using content ID (if a product category is selected)

Reviewer: ```suggestion
The Bestseller block is rendered based on content ID (if a product category is selected)
```
